packages/application-generic/src/usecases/add-job/add-digest-job.usecase.ts

```diff
@@ -72,6 +73,23 @@ export class AddDigestJob {
     if (matched > 0 && modified === 0) {
       await this.digestMergedExecutionDetails(job);
 
+      await this.jobRepository.update(
+        {
+          _environmentId: job._environmentId,
+          _id: job._id,
+        },
+        {
+          $set: {
+            status: JobStatusEnum.MERGED,
+          },
+        }
+      );
+
+      await this.jobRepository.updateAllChildJobStatus(
+        job,
+        JobStatusEnum.MERGED
+      );
```

Comment:
This should happen before `digestMergedExecutionDetails` as it would be misleading that in Execution Details we have that update and for any reason these 2 calls failed.
Also have you considered keep these 2 updates together in a single functionality? Where we could have another functionality that would need update job status from a parent job, for a status different to merged?